tests/pdata-test.cpp

```diff
@@ -1,10 +1,12 @@
 #include <gtest/gtest.h>
 
-extern "C" void pdata_test();
+extern "C" int count_stack_frames();
 
-TEST(Aarch64MinGW, PDATATest) 
+// This test asserts the number stack frames that can be unwound using RtlVirtualUnwind.
+// This is a good proxy to determine if pdata entries are correct.
+TEST(Aarch64MinGW, TestUnwindStack) 
 {
-    pdata_test();
+    EXPECT_EQ(count_stack_frames(), 22);
```

Comment:
I suggest to do all checks in C implementation it returns true or false for test result

Reply:
I think that including the stack frame count (and other results) in the assert would help the tests to be more diagnostic.
Returning true or false from C code would mean extra debugging steps for developers in future.

Reply:
I understand, it will be great to use asserts. However C++ is required here as gtest wrapper for C. Test can have many checks inside and it does not return enough data to make check in C++ wrapper. It is limitation of the toolchain. I prefer to have all check in C test for clarity. Diagnostic information  can be added by printf or by using some C wrappers.

Reply:
This PR is getting large. Will follow-up in next PR to avoid merging issues.